### ISSUE TITLE
DateInRegion ISO8601Formatter should be also able to create a DateInRegion in the specified timezone

### ISSUE THREAD

**Issue by phatmann** [malcommac/SwiftDate#223]
When using `DateInRegion(fromString:format)` to parse a date with a time zone (e.g. ISO8601Type.Full), the resulting DateInRegion does not use the parsed time zone, but instead uses the default time zone. This makes subsequent date calculations work incorrectly.

```
let dateInRegion = DateInRegion(fromString: 2016-07-18T08:00:00-06:00", format: .ISO8601Format)
print(dateInRegion)
```

Shows this output:

```
July 18, 2016 at 9:00:00 PM GMT+8
Calendar: gregorian
Time zone: Asia/Manila
Locale: en_US_POSIX
```

While it should be:

```
July 18, 2016 at 8:00:00 AM GMT-6
Calendar: gregorian
Time zone: America/LosAngeles
Locale: en_US_POSIX
```

I am willing to do a PR unless this behavior is somehow by design.

**Comment by malcommac** [comment]
Date is parsed from given timezone (-6h) to the absolute datetime format; the timezone of the output (dateInRegion) is defined by passed region parameter (by default, if not specified, as in your case, the timezone is your current locale timezone, so +8h).

**Comment by phatmann** [comment]
I understand the behavior and I have looked at the code. My point is that this behavior is unexpected. A string with a timezone should result in a DateInRegion with that same timezone. We should add code to parse the timezone and use that timezone if no region is provided.

**Comment by malcommac** [comment]
I've added it as enhancement.

**Comment by malcommac** [comment]
Since 4.1.0 `ISO8601Parser` supports parsing ISO8601 strings by evaluating automatically format's variant. It also take care of the specified timezone by creating `Date` or `DateInRegion` instance which reflect the value expressed in string.
Thank you for your request.

This is the reference commit: https://github.com/malcommac/SwiftDate/commit/255e90a95b7e31991d77256f98aafe651015ebc9

**Comment by phatmann** [comment]
@malcommac Great to see the new ISO parsing, nice work. However, a DateInRegion created via a ISO8601 string still does not have a region that reflects the timezone from the string. In fact, the region is always forced to GMT. So this issue should be re-opened.

From `init?(string: String, format: DateFormat, fromRegion region: Region? = nil)`:
```
...
let configuration = ISO8601Configuration(calendar: srcRegion.calendar)
guard let date = try ISO8601Parser(string, config: configuration).parsedDate else {
    return nil
}
self.absoluteDate = date
if srcRegion != Region.GMT() { // region is ignored
    print("Region is read from the string when ISO8601 parser is used")
}
srcRegion = Region.GMT()
...
```

Note that the timezone from the `ISO8601Parser` is not used to set the region.

**Comment by malcommac** [comment]
Thank you @phatmann , just fixed for 4.1.1 in https://github.com/malcommac/SwiftDate/commit/4bcd67ad4d3e0229ac717e1dddd2b5eae6ad3067
